dissect/ffs/ffs.py

```diff
@@ -219,6 +228,18 @@ def link(self):
 
         return self.open().read().decode("utf-8")
```

Comment:
```suggestion
        return self.open().read(self.size).decode("utf-8")
```
This should fix the null byte issue, so that you can remove the other change related to that.